### PR TITLE
Add a couple of commands and improve shift + note behaviour

### PR DESCRIPTION
**Pull request by jmckernon**
Details to follow in the Hackey Trackey forum thread (https://forum.cockos.com/showthread.php?t=202382).

### CODE REVIEW

File: Tracker/tracker.lua [hunk 1]
```diff
@@ -3462,61 +3477,65 @@ function tracker:createNote( inChar )
...
-        
+
         -- Were we overwriting an OFF marker?
         local idx = rows*chan+row
         if ( noteGrid[idx] and noteGrid[idx] < -1 ) then
           self:deleteNote(chan, row)
         end
       end
-      self.ypos = self.ypos + self.advance
+      if advance then
```
Comment (JoepVanlier) [inline Tracker/tracker.lua]: As far as I can tell, what you wrote will work, but requires if statements in a fair number of places, with the risk that in a future addition, I or someone else could forget one. I agree that the way I previously solved it (setting advance to zero outside this function was worse than this), but maybe an alternative option would be to make the function:

function tracker:createNote(inChar, advance_override)

and at the start set:
local advance = advance_override or self.advance

and then just use the local advance in the rest of this function. Then advance can be set to zero if you wish to override it and omitted if not. What do you think?

File: Tracker/tracker.lua [hunk 2]
```diff
@@ -7238,21 +7287,23 @@ function tracker:gotoCurrentPosition()
...
-      local oldAdvance = tracker.advance
-      tracker.advance = 0
-      tracker:createNote(lastChar+32)
+
+    if shift > 0 and string.match(string.char(lastChar),"[^%w]") == nil then
+      if not tracker.shiftChordInProgress then
+        tracker.shiftChordInProgress = true
+        tracker.shiftChordStartXpos = tracker.xpos
+      end
+      tracker:createNote(lastChar + 32, false)
```
Comment (JoepVanlier) [inline Tracker/tracker.lua]: Here it would then be tracker:createNote(lastchar + 32, 0)

File: Tracker/tracker.lua [hunk 3]
```diff
@@ -7238,21 +7287,23 @@ function tracker:gotoCurrentPosition()
...
       tracker:tab()
-      tracker.advance = oldAdvance
     else
-      tracker:createNote(lastChar)
+      tracker.shiftChordInProgress = false
+      tracker:createNote(lastChar, true)
```
Comment (JoepVanlier) [inline Tracker/tracker.lua]: And just tracker:createNote(lastChar + 32)

**Comment by JoepVanlier** [comment]
This slipped my mind earlier, but one minor thing that may be worth considering. Could be nice to make the new functionality optional. Could do this via:

Under -- Default configuration add
tracker.cfg.switchback = 1

Under "tracker.binaryOptions ="
add
    { 'switchback', 'Go back to first column when entering chords' },

And then change
if tracker.shiftChordInProgress and gfx.mouse_cap & 8 == 0 then
to
if tracker.cfg.switchback == 1 and tracker.shiftChordInProgress and gfx.mouse_cap & 8 == 0 then

Then it just becomes a toggle-able option in the options screen.

**Comment by jmckernon** [comment]
I agree that all the extra if statements are a bit nasty. I have an idea for a nicer way to do it, inspired by your suggestion but slightly different. I'll write it up in a bit and you can see whether you prefer it. If you don't like it then I'll implement your suggestion as described.

Happy to add it as an option, though I think it makes sense for it to be on by default. (I would be surprised if anyone preferred the old behaviour.)

**Comment by JoepVanlier** [comment]
Looks good to go! :+1: 

Thanks!

**Pull request merged**